### PR TITLE
[Silabs]Update Submodules, API, gn files to use latest silabs sdks

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
- Replace gecko sdk by simplicity sdk which is our new SDK
- Bump wifi sdk to 3.3.0
- Update matter support for new sdk support files and libs
- Update pulled files and configs in gn files structure
- Update some API that changed
- Use silabs new memory manager instead of freeRTOS heap4
- Use new silabs docker image in ci that utilise latest sdks

Tested various app combination builds, commissioning and control on EFR32MG24
CI will also test a series of combinations

### CODE REVIEW

File: third_party/silabs/efr32_sdk.gni [hunk 1]
```diff
@@ -99,6 +96,9 @@ declare_args() {
   sl_ot_platform_abstraction =
       "${efr32_sdk_root}/protocol/openthread/platform-abstraction"
 
+  # Enable Synchronized Sleepy End Device
+  enable_synchronized_sed = use_silabs_thread_lib
```
Comment (mkardous-silabs) [inline third_party/silabs/efr32_sdk.gni:100]: Why is this change necessary?

**Review by mkardous-silabs — APPROVED** [review]

File: third_party/silabs/efr32_sdk.gni (hunk 1 above)
Comment (jmartinez-silabs) [reply third_party/silabs/efr32_sdk.gni]: fixed

**Review by yufengwangca — APPROVED** [review]
